### ISSUE TITLE
Do not assume my namespace belongs to you

### ISSUE THREAD

**Issue by sholladay** [jrburke/r.js#843]
## Problem
My application (we'll call it `banana`, for sake of discussion), lives on a namespace. `window.banana`, for example. When using the r.js optimizer with the `namespace: 'banana'` setting, and `include`ing [Alameda](https://github.com/requirejs/alameda) as [I do here](https://github.com/jrburke/r.js/issues/842), the optimized file ends up having something like ...
````javascript
if (banana) {
   require = banana;
}
````
... and then Alameda proceeds to treat `require` as its own configuration, since it was already defined.

In other words, you are treating my namespace as the loader config. And that's just no good for me. See my use case in #842 for more info.

**Comment by sholladay** [comment]
I am thinking that r.js is doing a little too much work here automatically. While a config option could be added that disables this bad behavior, I think it makes more sense to push responsibility of this setup code to the user (via something akin to `wrap`), if they are going to use a namespace. Then we don't have to hack around various assumptions like this. To make it easier, there could simply be a default for the `wrap`-like option that does exactly what the existing code does.
